src/senselab/audio/tasks/forced_alignment/forced_alignment.py

```diff
@@ -0,0 +1,743 @@
+        data (AlignedTranscriptionResult): The input dictionary with segments and word segments.
+
+    Returns:
+        List[ScriptLine]: The list of ScriptLine objects.
+    """
+    segments = data["segments"]
+    script_lines = []
+
+    for segment in segments:
+        words = segment["words"]
+        word_chunks = [ScriptLine(text=word["word"]) for word in words]
+
+        # Handle 'nan' end values by setting them to None
+        start = segment["start"]
+        end: Optional[float] = segment["end"]
+        if end is not None and (isinstance(end, float) and math.isnan(end)):
+            end = None
+
+        script_line = ScriptLine(text=segment["text"], start=start, end=end, chunks=word_chunks)
+        script_lines.append(script_line)
+
+    return script_lines
+
+
+def _align_transcription(
+    transcript: List[SingleSegment],
+    model: torch.nn.Module,
+    align_model_metadata: Dict[str, Any],
+    audio: Audio,
+    device: str,
+    interpolate_method: str = "nearest",
+    return_char_alignments: bool = False,
+    print_progress: bool = False,
+    combined_progress: bool = False,
+) -> AlignedTranscriptionResult:
+    """Aligns phoneme recognition predictions to known transcription.
+
+    Args:
+        transcript (List[SingleSegment]): The list of transcription segments.
+        model (torch.nn.Module): The alignment model.
+        align_model_metadata (Dict[str, Any]): Metadata for the alignment model.
+        audio (np.ndarray): The audio data.
+        device (str): The device to run the model on.
+        interpolate_method (str): The method for interpolating NaNs (default: "nearest").
+        return_char_alignments (bool): Whether to return character alignments (default: False).
+        print_progress (bool): Whether to print progress (default: False).
+        combined_progress (bool): Whether to combine progress (default: False).
+
+    Returns:
+        AlignedTranscriptionResult: The aligned transcription result.
+    """
+    audio = _prepare_audio(audio)
+    max_duration = audio.waveform.shape[1] / audio.sampling_rate
+
+    model_dictionary = align_model_metadata["dictionary"]
+    model_lang = align_model_metadata["language"]
+    model_type = align_model_metadata["type"]
+
+    transcript = _preprocess_segments(
+        transcript,
+        align_model_metadata["dictionary"],
+        align_model_metadata["language"],
+        print_progress,
+        combined_progress,
+    )
+
+    aligned_segments, word_segments = _align_segments(
+        transcript=transcript,
+        model=model,
+        model_dictionary=model_dictionary,
+        model_lang=model_lang,
+        model_type=model_type,
+        audio=audio,
+        device=device,
+        max_duration=max_duration,
+        return_char_alignments=return_char_alignments,
+        interpolate_method=interpolate_method,
+    )
+    return {"segments": aligned_segments, "word_segments": word_segments}
+
+
+def align_transcriptions(
```

Comment:
@ibevers what if the audio clips (and not audios lol) and the transcripts are not in the same number?